### DISCUSSION TITLE
Contributing new intents?

### DISCUSSION THREAD

**Discussion by petercwallis** [OHF-Voice/intents#212] [_General]
Hi all. I am excited by the possibilities of HomeAssistant's year of voice.  I was Principle Investigator on a EU project that put "talking rabbits" (nabaztag) in older peoples homes to encourage exercise. We had PIR sensors and knew where the house keys were.  This was back in 2009-2012 so before the Echo and we had major issues with the speech recognition I must say.  Things have moved on in the last 10 years but the "understanding" side of voice interfaces does appear to be stuck.  I can help.. I hope.

The intents for home assistant are very much about "control" of the home, which is fine when the user a) knows what he/she is doing, and b) does only good things. An alternative is that a voice interface is, and acts as, a social actor, and I think this would be interesting to many in the community.  To this end I was wondering how I should go about suggesting two more (sets of) "intents".

The first is based on being helpful.  The gist is that "turn on the bedroom light" should not simply say that "bedroom light" was not found, but should say what it can turn on, and do that _not_ as a long list, but as part of an ongoing dialog.  For instance listing the 3 most popular lights. An "intent" HassListOptions (say) would have a field specifying the HassTurnOn intent, and the action would be complete when the user turn(ed)on something or cancelled somehow.

A second is based around authority.  For cloud-based solutions identifying individuals is problematic but for a local service the technology is the limit.  Assuming the audio signal processing automatically assigns a probable id to a voice, Questions such as "Are you allowed to open the garrage door?" make sense.  Any child worth his or her salt will say "Yes", but the system is then justified in asking a "favourite colour" quesiton or indeed ask for a password.  The intent might be HassCheckAuth or something.

The big picture is that language in use is actually an action in a social space, and hence could do with an "intent" to make that explicit.  In 2023 I intend to set up home Assistant with a voice interface so that I can play with the voice dimension.  I am new to git hub and am not confident about how this whole community process works.  I get the impression I should fork the project, and work with that.  Please let me know if that is not right - I imagine I need to lodge pull requests in incremental bits but am not sure how that would work.

**Comment by balloob** [comment]
Right now we want to limit to basic interactions because we're still working on building out the very basic foundation of the platform. We might explore such features, but I don't see that happening in 2023.

If people are interested in this, Home Assistant's APIs are open and our data sets available, so people can pursue that.

**Reply by petercwallis** [reply]
Okay. So I should fork and play.  Fine, but the strength of your approach to voice is (I think) that multiple languages are integrated from the start.  Could I propose that there is some kind of 2D matrix of intents vs language.  Anyone with a language skill can add a language, but adding an intent is more problematic because it requires others to fill out the languages dimension. I will think about it ...

**Reply by fadamsen** [reply]
Hi @petercwallis. Regarding the 2D matrix, is [this](https://home-assistant.github.io/intents/) along the lines of what you are looking for?

**Comment by petercwallis** [comment]
Yep.  So adding a row is easy; and what needs a process is for us to add a
column. That is tricky.

**Comment by drmaddy736** [comment]
Marathi language leader?
